next.clayui.com/content/docs/foundations/composing.md

```diff
@@ -3,4 +3,146 @@ title: "Composing"
+
+## Composition
+
+The composition is very strong with React, JSX syntactic sugar makes it very powerful and more viable when coupled with special `children` prop for passing arbitrary child elements. Props in React components allows us to pass objects and consequently React elements since they are objects.
+
+```jsx
+<ClayModal>
+	<ClayModal.Footer
+		first={
+			<ClayButton.Group spaced>
+				<ClayButton displayType="secondary">
+					{'Secondary'}
+				</ClayButton>
+				<ClayButton displayType="secondary">
+					{'Secondary'}
+				</ClayButton>
+			</ClayButton.Group>
+		}
+		middle={
+			<ClayButton>{'Primary'}</ClayButton>
+		}
+	/>
+</ClayModal>
+```
+
+So we use some of the composable component approach in React.
+
+<div class="clay-ullist">
+
+- [**Containment**](https://reactjs.org/docs/composition-vs-inheritance.html#containment) - They are treated as generic blocks that accept the passing of arbitrary `children` to them, either using `children` props or the special naming convention itself.
+- [**Specialization**](https://reactjs.org/docs/composition-vs-inheritance.html#specialization) - "Special case" components that are specific for some use purposes and not treated as something more "generic".
+
+</div>
+
+## Philosophy
+
+Assuming that we want our components to be reusable and enable developers to have the flexibility to build on their use cases while helping developers who want to do something fast when just following [Lexicon](https://lexicondesign.io) specifications, we follow a few Thought lines.
+
+Teams are constantly advancing and new features are always emerging that require new interactions and behavioral changes, a product changes a lot over time, being refined, aligning interests and changing behaviors in the UI.
+
+Designers cannot get caught up in strict rules this blocks the creative mind and can prevent UI advances and interaction enhancements. The same goes for components, it has to allow developers to move forward with the standards made by designers and allow them to reuse components in the future.
+
+Dealing with this is complex and requires balancing component flexibility and specialization, getting to that balance is difficult and we can spend a lot of time polishing and refining components to achieve **"component supremacy"**.
+
+We are delivering Clay as a **Multi-Layered API library**: **low-level**, **high-level** and **pure html**. Each approach has its purposes of use:
+
+<div class="clay-ullist">
+
+- **low-level** - Flexibility and use of first level composition so you can customize and create high level components.
+- **high-level** - Speed ​​to build new features are usually specialized components and tend to cover only a few use cases, limiting their flexibility.
+
```

Comment:
what about pure html? Im guessing this should just point to clay-css? Or we may not even need to include it since  this doc is specific to composition

Reply:
Hmm, not sure, maybe Patrick could work on a similar but specific document for clay-css.